test/rules/prefer-for-of/test.ts.lint

```diff
@@ -178,4 +178,20 @@ function test() {
         print();
 }
 
+for (let i = 0; i < arr.length; ++i) {
+    arr[i]++;
+}
+
+for (let i = 0; i < arr.length; ++i) {
+    ({...arr[i]} = foo);
+}
+
+for (let i = 0; i < arr.length; ++i) {
+    for (arr[i] of otherArr) {}
```

Comment:
Didn't even realize that this would be valid - I hope nobody ever does this in real life :P